testing/endtoend/components/eth1/depositor.go

```diff
@@ -0,0 +1,208 @@
+package eth1
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/prysmaticlabs/prysm/v3/encoding/bytesutil"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	contracts "github.com/prysmaticlabs/prysm/v3/contracts/deposit"
+	eth "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	e2e "github.com/prysmaticlabs/prysm/v3/testing/endtoend/params"
+	"github.com/prysmaticlabs/prysm/v3/testing/endtoend/types"
+	"github.com/prysmaticlabs/prysm/v3/testing/util"
+)
+
+const depositGasLimit = 4000000
```

Comment:
Any commentary / context you can add in the code about this value?

Reply:
This was refactored from an identical uncommented constant, so I have no idea how we landed at this particular number. Considering all transactions in e2e come from this code, and therefore use the same limit, I think it's probably pretty arbitrary.

Reply:
@nisdas by chance do you have any helpful commentary to add to this constant?

Reply:
Oh I forgot about the spam tx to advance the chain (in `WaitForBlocks`), which use a gas limit of `21000`. IDK if the relationship between these numbers is meaningful (other than the spam tx being a few orders of magnitude smaller than the deposit tx).